packages/esm-patient-chart-app/src/visit/widget/visits-overview.component.tsx

```diff
@@ -0,0 +1,238 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import dayjs from 'dayjs';
+import styles from './visits-overview.scss';
+import Button from 'carbon-components-react/es/components/Button';
+import DataTableSkeleton from 'carbon-components-react/es/components/DataTableSkeleton';
+import DataTable, {
+  Table,
+  TableCell,
+  TableContainer,
+  TableBody,
+  TableHead,
+  TableHeader,
+  TableRow,
+  TableExpandRow,
+  TableExpandedRow,
+  TableExpandHeader,
+} from 'carbon-components-react/es/components/DataTable';
+import { useTranslation } from 'react-i18next';
+import { EmptyState, ErrorState } from '@openmrs/esm-patient-common-lib';
+import { Visit, getVisitsForPatient, createErrorHandler, OpenmrsResource } from '@openmrs/esm-framework';
+import { fetchEncounterByUuid } from './encounter.resource';
+import { StructuredListBody } from 'carbon-components-react';
+
+const visitsToShowCount = 5;
+
+interface EncounterViewProps {
+  encounter: OpenmrsResource;
+}
+
+const RenderEncounter: React.FC<EncounterViewProps> = ({ encounter }) => {
+  const { t } = useTranslation();
+  const [encounterDetails, setEncounterDetails] = useState<OpenmrsResource | null>(null);
+  const { uuid, encounterType } = encounter;
+
+  const formatVisitTime = (inputDate: any) => {
+    return dayjs(inputDate).format('HH:MM');
+  };
+
+  useEffect(() => {
+    if (uuid) {
+      const sub = fetchEncounterByUuid(uuid).subscribe(setEncounterDetails, createErrorHandler());
+      return () => sub.unsubscribe();
+    }
+  }, [uuid]);
+  console.log(encounterDetails, encounterType.display, 'encounterDetails');
+
+  return (
+    <div className={styles.encounterRow}>
+      <p className={styles.leftSide}>{encounterType.display}</p>
+      <div className={styles.rightSide}>
+        {encounterDetails && (
+          <React.Fragment>
+            {encounterDetails.obs.length > 0 &&
+              encounterDetails.obs.map((ob, index) => <p key={index}>{ob.display}</p>)}
+            <p>Time: {formatVisitTime(encounterDetails.encounterDatetime)}</p>
+          </React.Fragment>
+        )}
+      </div>
+    </div>
+  );
+};
+
+interface RenderVisitsProps {
+  headerTitle: string;
+  visits: Array<Visit>;
+  tableRows: Array<{
+    id: string;
+    date: string;
+    time: string;
+    location: string;
+    visitType: string;
+    encounters: Array<OpenmrsResource>;
+  }>;
+  showAllVisits: boolean;
+  toggleShowAllVisits(): void;
+}
+
+const RenderVisits: React.FC<RenderVisitsProps> = ({
+  headerTitle,
+  visits,
+  tableRows,
+  showAllVisits,
+  toggleShowAllVisits,
+}) => {
+  const { t } = useTranslation();
+  const displayText = t('startVisit', 'Visit');
+
+  const tableHeaders = [
+    { key: 'date', header: 'Date' },
+    { key: 'time', header: 'Time' },
+    { key: 'location', header: 'Location' },
+    { key: 'visitType', header: 'Visit Type' },
+  ];
+
+  const launchStartVisitPrompt = React.useCallback(
```

Comment:
Honestly, the whole thing can be made a "constant" function by moving it out of the component.

There is no need to re-allocate this on every render.